src/mavis/convert/vcf.py

```diff
@@ -122,6 +123,99 @@ def parse_bnd_alt(alt: str) -> Tuple[str, int, str, str, str, str]:
+                COLUMNS.break2_position_end: bp_end + record.info.get('CILEN', (0, 0))[1],
+            }
+        )
+    else:
+        std_row.update(
+            {
+                COLUMNS.break1_position_start: max(
+                    1, record.pos + record.info.get('CIPOS', (0, 0))[0]
+                ),
+                COLUMNS.break1_position_end: record.pos + record.info.get('CIPOS', (0, 0))[1],
+                COLUMNS.break2_position_start: max(1, bp_end + record.info.get('CIEND', (0, 0))[0]),
+                COLUMNS.break2_position_end: bp_end + record.info.get('CIEND', (0, 0))[1],
+            }
+        )
+
+    if std_row["break1_chromosome"] == std_row["break2_chromosome"] and (
+        Interval.overlaps(
+            (std_row['break1_position_start'], std_row['break1_position_end']),
+            (std_row['break2_position_start'], std_row['break2_position_end']),
+        )
+    ):
+        if 'event_type' in std_row and std_row['event_type'] != 'BND':
+            std_row['break2_position_start'] = max(
+                std_row['break1_position_start'], std_row['break2_position_start']
+            )
+            std_row['break1_position_end'] = min(
+                std_row['break1_position_end'], std_row['break2_position_end']
+            )
+            std_row['break1_position_start'] = min(
+                std_row['break1_position_start'], std_row['break2_position_end']
+            )
+    if std_row['break1_position_end'] == 0 and std_row['break1_position_start'] == 1:
+        # addresses cases where pos = 0 and telomeric BND alt syntax https://github.com/bcgsc/mavis/issues/294
+        std_row.update({'break1_position_end': 1})
+    if std_row['break2_position_end'] == 0 and std_row['break2_position_start'] == 1:
+        std_row.update({'break2_position_end': 1})
+
+    if std_row["break1_chromosome"] == std_row["break2_chromosome"] and (
+        std_row['break2_position_start'] > std_row['break2_position_end']
+        or std_row['break1_position_start'] > std_row['break1_position_end']
+    ):
+        if 'event_type' in std_row and std_row['event_type'] != 'BND':
+            logger.error(
+                f'Improper entry. One of the following breakpoints start is greater than breakpoint end:\n Breakpoint1_start: {std_row["break1_position_start"]}, Breakpoint1_end: {std_row["break1_position_end"]}\n Breakpoint2_start: {std_row["break2_position_start"]}, Breakpoint2_end: {std_row["break2_position_end"]}\n This call has been dropped.'
+            )
+            std_row.clear()
+
+    if (
+        not None in (record.pos, record.info.get('END'))
+        and record.pos > record.info.get('END')
+        and std_row["break1_chromosome"] == std_row["break2_chromosome"]
+    ):
+        logger.error(
+            f'Improper entry. Starting position ({record.pos}) cannot be greater than ending position ({record.info.get("END")}).\n This call has been dropped.'
+        )
+        std_row.clear()
```

Comment:
rather than doing a `clear` it would make more sense to throw an error in this function and catch it in the parent function